@xen-orchestra/lite/src/views/vm/VmDashboardView.vue

```diff
@@ -1,11 +1,120 @@
 <template>
-  <PageUnderConstruction />
+  <VtsObjectNotFoundHero v-if="vm === undefined" :id type="page" />
+  <div v-else class="vm-dashboard-view" :class="{ mobile: uiStore.isMobile }">
+    <VmDashboardQuickInfo class="quick-info" :vm />
+    <div v-if="data.stats === undefined" class="offline-hero-container">
+      <VtsOfflineHero type="page" />
+    </div>
+    <template v-else>
+      <VmDashboardCpuUsageChart class="cpu-usage-chart" :data :error="lastError" :loading="isFetching" />
+      <VmDashboardRamUsageChart class="ram-usage-chart" :data :error="lastError" :loading="isFetching" />
+      <VmDashboardNetworkUsageChart class="network-usage-chart" :data :error="lastError" :loading="isFetching" />
+      <VmDashboardVdiUsageChart class="disk-usage-chart" :data :error="lastError" :loading="isFetching" />
+    </template>
+  </div>
 </template>
 
 <script lang="ts" setup>
-import PageUnderConstruction from '@/components/PageUnderConstruction.vue'
+import VmDashboardCpuUsageChart from '@/components/vm/dashboard/VmDashboardCpuUsageChart.vue'
+import VmDashboardNetworkUsageChart from '@/components/vm/dashboard/VmDashboardNetworkUsageChart.vue'
+import VmDashboardQuickInfo from '@/components/vm/dashboard/VmDashboardQuickInfo.vue'
+import VmDashboardRamUsageChart from '@/components/vm/dashboard/VmDashboardRamUsageChart.vue'
+import VmDashboardVdiUsageChart from '@/components/vm/dashboard/VmDashboardVdiUsageChart.vue'
+import useFetchStats from '@/composables/fetch-stats.composable.ts'
+import { GRANULARITY } from '@/libs/xapi-stats.ts'
+import type { XenApiVm } from '@/libs/xen-api/xen-api.types.ts'
 import { usePageTitleStore } from '@/stores/page-title.store'
+import { useVmStore } from '@/stores/xen-api/vm.store.ts'
+import VtsObjectNotFoundHero from '@core/components/state-hero/VtsObjectNotFoundHero.vue'
+import VtsOfflineHero from '@core/components/state-hero/VtsOfflineHero.vue'
+import { useUiStore } from '@core/stores/ui.store.ts'
+import { computed, onUnmounted, watchEffect } from 'vue'
 import { useI18n } from 'vue-i18n'
+import { useRoute } from 'vue-router'
 
-usePageTitleStore().setTitle(useI18n().t('dashboard'))
+const { t } = useI18n()
+
+usePageTitleStore().setTitle(t('dashboard'))
+
+const route = useRoute()
+
+const id = computed(() => route.params.uuid as XenApiVm['uuid'])
+
+const { getByUuid, getStats: getVmStats, isFetching, lastError } = useVmStore().subscribe()
+
+const vm = computed(() => getByUuid(id.value))
+
+const uiStore = useUiStore()
+
+const { stats, register, unregister, timestampStart } = useFetchStats<XenApiVm>(getVmStats, GRANULARITY.Hours)
```

Comment:
The return type is already the right one, without having to pass the generic:
```suggestion
const { stats, register, unregister, timestampStart } = useFetchStats(getVmStats, GRANULARITY.Hours)
```